packages/react-native/ReactAndroid/src/main/java/com/facebook/react/views/view/WindowUtil.kt

```diff
@@ -67,39 +67,39 @@ private fun Window.statusBarShow() {
 }
 
 @Suppress("DEPRECATION")
-public fun Window.setSystemBarsTranslucency(isTranslucent: Boolean) {
-  WindowCompat.setDecorFitsSystemWindows(this, !isTranslucent)
+public fun Window.enableEdgeToEdge() {
+  val isDarkMode = ContextUtils.isDarkMode(context)
 
-  if (isTranslucent) {
-    val isDarkMode =
-        context.resources.configuration.uiMode and Configuration.UI_MODE_NIGHT_MASK ==
-            Configuration.UI_MODE_NIGHT_YES
-
-    if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
-      isStatusBarContrastEnforced = false
-      isNavigationBarContrastEnforced = true
-    }
+  WindowCompat.setDecorFitsSystemWindows(this, false)
 
-    statusBarColor = Color.TRANSPARENT
-    navigationBarColor =
-        when {
-          Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q -> Color.TRANSPARENT
-          Build.VERSION.SDK_INT >= Build.VERSION_CODES.O_MR1 && !isDarkMode ->
-              Color.argb(0xe6, 0xFF, 0xFF, 0xFF)
-          else -> Color.argb(0x80, 0x1b, 0x1b, 0x1b)
-        }
+  if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.Q) {
+    isStatusBarContrastEnforced = false
```

Comment:
Also just a random thought in my head...

Would it be possible to expose a new component (`NavigationBar`, similar to `StatusBar`) and add a possibility to customize properties of NavigationBar (backgroundColor/translucent/style (dark/light))?

I know that there are libraries `react-native-bars` that were aiming to deliver this API. But since we add an ability to toggle `edge-to-edge` mode, maybe it would be reasonable to add other components as well, so that RN will have a full feature set of component out of the box?

Reply:
For me, having a `NavigationBar` component is a bit out-of-scope as a lot of these props will have no effect. `react-native-edge-to-edge` has a `SystemBars` component, which doesn't uses deprecated APIs and align to what is enforced on Android 15. And only `hidden` matters for navigation bar.

- `backgroundColor` -> will have no effect with gesture navigation, only with buttons navigation
- `barStyle` -> same, as gesture navigation style is automatic and based on contrast

For those two to be usable, we also need to set `[setStatusBarContrastEnforced](https://developer.android.com/reference/android/view/Window#setStatusBarContrastEnforced(boolean))` to `false`, which doesn't align with what's enforced in Android 15 / what occurs when AndroidX core`enableEdgeToEdge` is used.

They keep a semi opaque nav bar (white or black, dependending on the color scheme) for buttons nav, transparent for gesture nav. That avoids the additional burden to handle nav bar color / style only for Android users with buttons navigation on (not even the default). I think it's not worth the extra work, and the Google choice is probably the good one here.